src/core/StelLocationMgr.cpp

```diff
@@ -747,12 +747,20 @@ const StelLocation StelLocationMgr::locationFromCLI() const
 {
 	StelLocation ret;
 	QSettings* conf = StelApp::getInstance().getSettings();
-	bool ok;
 	conf->beginGroup("location_run_once");
-	ret.latitude = parseAngle(StelUtils::radToDmsStr(conf->value("latitude").toDouble(), true), &ok);
-	if (!ok) ret.role = '!';
-	ret.longitude = parseAngle(StelUtils::radToDmsStr(conf->value("longitude").toDouble(), true), &ok);
-	if (!ok) ret.role = '!';
+
+	const auto latVar = conf->value("latitude");
+	if (latVar.isValid())
+		ret.latitude = 180/M_PI * latVar.toDouble();
```

Comment:
OK, this is the minimal change that removes this nonsense, but maybe does not yet fix the actual problem. Maybe in CLIProcessor the first translation should go to  decimal degrees already?

Reply:
Or do you mean by "problem" the fact that the value is saved in radians instead of degrees?

Reply:
But does it also work with DMS formatted ones?

Reply:
As commented in the main thread, yes it does. I guess you are puzzled why this change fixes the problem. The actual problem is in the [parseAngle](https://github.com/Stellarium/stellarium/blob/53299955f3448a4f17cb9338eea975cecaabd58c/src/core/StelLocationMgr.cpp#L675) function in the same file. This should indeed be fixed so that parsing of locations works correctly, but this patch is a separate change, which, incidentally, does fix #2239.

Reply:
d'oh, OK. Then parseAngle is and remains broken and must still be fixed. Either in a new issue/PR or in this one.

Reply:
Yeah, I was going to make another PR for this (don't like my diligently separated commits being squashed).